request-cert/main.go

```diff
@@ -52,53 +51,103 @@ func main() {
 
 	// Check certificate type.
 	var template *x509.CertificateRequest
-	var filePrefix string
+	var filename, csrName string
 	var wantServerAuth bool
 
+	hostname, err := os.Hostname()
```

Comment:
Using just `node.<hostname>` won't lead to unique CSRs if someone creates multiple cockroach clusters within one k8s cluster, spread across multiple namespaces. That seems like it may be an issue since CSRs don't appear to be scoped to a namespace. The same issue appears possible for user certs as well.

Can we scope the CSR names with the namespace and cluster name? Or am I misunderstanding this?